tools/parameter_tuning.py

```diff
@@ -0,0 +1,56 @@
+from sisyphus import Job, Task, tk
+from typing import Any, Literal, Sequence, Union
+
+import numpy as np
+
+from i6_core.util import instanciate_delayed
+
+
+class GetOptimalParametersAsVariableJob(Job):
+    """
+    Pick a set of optimal parameters based on their assigned (dynamic) score value.
+    Each optimal parameter is outputted individually to be accessible in the Sisyphus manager.
+
+    Can be used to e.g. pick best lm-scale and prior scale to a corresponding ScliteJob.out_wer.
+    """
+
+    def __init__(
+        self,
+        *,
+        parameters: Sequence[Sequence[Any]],
+        values: Sequence[tk.Variable],
+        mode: Union[Literal["maximize"], Literal["minimize"]],
+    ):
+        """
+        :param parameters: a row / column matrix of parameters, where the best row should be selected
+        :param values: sequence of tk.Variables with the same length as row, containing int or float.
+            Used to determine the best row of parameters.
+            Some calculations might be done using DelayedOps math beforehand.
```

Comment:
I think calling this a matrix is confusing. Also, it does not explain my aspect why it must be a sequence.
```suggestion
        :param parameters: parameters[best_idx] will be written to self.out_optimal_parameters
           as Sisyphus output variables.
           parameters[best_idx] (and thus self.out_optimal_parameters) is a sequence of fixed length,
           to allow to index into it.
           Thus, len(parameters[i]) must be the same for all i.
        :param values: best_idx = argmax(values) or argmin(values).
            Must have len(values) == len(parameters).
            Some calculations might be done using DelayedOps math beforehand.
```

Reply:
For me this docstring is not really helpful, but I also do not want to delay so I will just commit